packages/integrations/src/openmediavault/openmediavault-integration.ts

```diff
@@ -0,0 +1,149 @@
+import { Integration } from "../base/integration";
+import { IntegrationTestConnectionError } from "../base/test-connection-error";
+import type { HealthMonitoring } from "../types";
+import { cpuTempSchema, fileSystemSchema, smartSchema, systemInformationSchema } from "./openmediavault-types";
+
+export class OpenMediaVaultIntegration extends Integration {
+  private username = this.getSecretValue("username");
+  private password = this.getSecretValue("password");
+  private async makeOpenMediaVaultRPCCallAsync(
+    serviceName: string,
+    method: string,
+    params: Record<string, unknown>,
+    headers: Record<string, string> = {},
+  ): Promise<Response> {
+    const url = `${this.integration.url}/rpc.php`;
+    return await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        ...headers,
+      },
+      body: JSON.stringify({
+        service: serviceName,
+        method,
+        params,
+      }),
+    });
+  }
+
+  public async testConnectionAsync(): Promise<void> {
+    const response = await this.makeOpenMediaVaultRPCCallAsync("session", "login", {
+      username: this.username,
+      password: this.password,
+    });
+
+    if (!response.ok) {
+      throw new IntegrationTestConnectionError("invalidCredentials");
+    }
+    const result = (await response.json()) as unknown;
+    if (!(typeof result === "object" && result !== null && "response" in result)) {
```

Comment:
``typeof result === "object"`` should make ``result !== null`` obsolete, no?

Reply:
No sadly not, `null` is an `object` in terms of typescript :joy:

Reply:
So the condition is OK?